src/core/server/saved_objects/types.ts

```diff
@@ -89,6 +89,13 @@ export interface SavedObjectsFindOptions {
   defaultSearchOperator?: 'AND' | 'OR';
   filter?: string | KueryNode;
   namespaces?: string[];
+  /**
+   * Map that is a subset of types and namespaces to search in; this is only intended to be used by a saved object client wrapper.
+   * If this is defined, it supersedes the `type` and `namespaces` fields when building the Elasticsearch query.
+   * Any types that are not included in this map will be excluded entirely.
+   * If a type is included but its value is undefined, the operation will search for that type in the Default namespace.
+   */
+  typesAndNamespacesMap?: Map<string, string[] | undefined>;
```

Comment:
NIT: Maybe that's just me, but I did not understand the name and description without looking at the implementation.

Not sure we have any naming convention for that, but I would rename that to `typeToNamespacesMap` which (imho) is a little more explicit. The tsdoc could probably also be slightly improved to explain that this is a `type -> namespace(s) to search for the type in` map.